src/autocomplete/databases/mongo/mongo-extract-commands.ts

```diff
@@ -0,0 +1,267 @@
+                parameters: formatJson5(childContext.skipModifierArgument().getText()),
+            };
+        case childContext instanceof LimitModifierContext:
+            return {
+                method: 'limit',
+                parameters: formatJson5(childContext.limitModifierArgument().getText()),
+            };
+        case childContext instanceof FilterModifierContext:
+            return {
+                method: 'filter',
+                parameters: formatJson5(childContext.filterModifierArgument().getText()),
+            };
+        case childContext instanceof MinModifierContext:
+            return {
+                method: 'min',
+                parameters: formatJson5(childContext.minModifierArgument().getText()),
+            };
+        case childContext instanceof MaxModifierContext:
+            return {
+                method: 'max',
+                parameters: formatJson5(childContext.maxModifierArgument().getText()),
+            };
+        case childContext instanceof ReturnKeyModifierContext:
+            return {
+                method: 'returnKey',
+                parameters: formatJson5(childContext.returnKeyModifierArgument().getText()),
+            };
+        case childContext instanceof ShowRecordIdModifierContext:
+            return {
+                method: 'showRecordId',
+                parameters: formatJson5(childContext.showRecordIdModifierArgument().getText()),
+            };
+        case childContext instanceof SortModifierContext:
+            return {
+                method: 'sort',
+                parameters: formatJson5(childContext.sortModifierArgument1().getText()),
+                options: formatJson5(childContext.sortModifierArgument2()?.getText()),
+            };
+        case childContext instanceof HintModifierContext:
+            return {
+                method: 'hint',
+                parameters: formatJson5(childContext.hintModifierArgument().getText()),
+            };
+        default:
+            throw newExpectedError('Modifier is not implemented: ' + childContext?.getText());
+    }
+}
+
+function formatJson5(string: string | undefined): undefined | unknown {
+    return string ? json5.parse<unknown>(string) : undefined;
+}
+
+export function extractMongoCommandsFromQuery(query: string): ExtractMongoCommandsFromQueryResult {
+    const parser = createParser(MongoLexer, MongoParser, query);
+
+    const commands: Command[] = [];
+    const onParseCommand: OnParseCommand = (command) => {
+        commands.push(command);
+    };
+
+    let parseCommandsError: ExpectedError | UnexpectedError | undefined;
+    const onParseCommandsError: OnParseCommandError = (error) => {
+        parseCommandsError = error;
+    };
+
+    const syntaxErrorListener = new SqlErrorListener(MongoParser.WS);
+    parser.addErrorListener(syntaxErrorListener);
+
+    const visitor = new CommandsVisitor(onParseCommand, onParseCommandsError);
+    getParseTree(parser).accept(visitor);
```

Comment:
is there a reason we're using `accept` instead of `visit` like here? https://github.com/gravity-ui/websql-autocomplete/blob/main/src/autocomplete/shared/tables.ts#L282

I don't know what the difference is

Reply:
this is method of parse tree, I will use visitor method for consistency